nixarr/monitoring/default.nix

```
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.nixarr;
+  
+  # Helper function to determine if a specific exporter should be enabled
+  shouldEnableExporter = service: 
+    cfg.${service}.enable && 
+    (cfg.monitoring.exporters.${service} or true);
+    
+  # Helper function to determine if a service is VPN-confined
+  isVpnConfined = service: cfg.${service}.enable && cfg.${service}.vpn.enable;
+    
+  # Helper function to create a script that extracts API key from config.xml
+  extractApiKeys = pkgs.writeShellApplication {
+    name = "extract-monitoring-api-keys";
+    runtimeInputs = with pkgs; [ dasel coreutils ];
+    text = ''
+      # Create state directory for API keys
+      STATE_DIR="/var/lib/exportarr"
+      mkdir -p "$STATE_DIR"
+      chmod 755 "$STATE_DIR"
```

```suggestion
      chmod 700 "$STATE_DIR"
```

For good measure. Or maybe `750` if we go with the `/data/.state/nixarr/api-keys` approach.